UE4SS/src/Mod/LuaMod.cpp

```diff
@@ -3762,25 +3763,35 @@ No overload found for function 'FPackageName:IsValidLongPackageName'.
+                                                                             m_static_construct_object_lua_callbacks.end(),
+                                                                             [&](LuaCancellableCallbackData& callback_data) -> bool {
+                                                                                 bool cancel = false;
+                                                                                 if (callback_data.instance_of_class == object_class)
+                                                                                 {
+                                                                                     try
+                                                                                     {
+                                                                                         callback_data.lua->registry().get_function_ref(callback_data.lua_callback_function_ref);
+                                                                                         LuaType::auto_construct_object(*callback_data.lua, constructed_object);
+                                                                                         callback_data.lua->call_function(1, 1);
+
+                                                                                         cancel = callback_data.lua->is_bool(-1) && callback_data.lua->get_bool(-1);
+                                                                                     }
+                                                                                     catch (std::runtime_error& e)
+                                                                                     {
+                                                                                         Output::send(STR("{}\n"), to_wstring(e.what()));
+                                                                                     }
+
+                                                                                     if (cancel)
+                                                                                     {
+                                                                                         // Release the thread_ref to GC.
+                                                                                         luaL_unref(callback_data.lua->get_lua_state(), LUA_REGISTRYINDEX, callback_data.lua_callback_thread_ref);
+                                                                                     }
+                                                                                 }
+
+                                                                                 return cancel;
+                                                                             }),
+                                                              m_static_construct_object_lua_callbacks.end());
```

Comment:
I haven't ran clang-format yes, although for both points, I've been replicating https://github.com/UE4SS-RE/RE-UE4SS/blob/332eea5711d036edd428b733f27214e8128e3d96/UE4SS/src/Mod/LuaMod.cpp#L4362-L4400 both in its massive indentation and `.erase` + `::remove_if` pattern.

Reply:
I'm not sure why clang-format thinks that the code you based your code on needs to be indented like that and not your code but I'm sure it has its reasons, and it's irrelevant since we're always suppose to run clang-format to make sure everyone uses the same style.

Also, it looks like `std::erase_if` should eliminate a lot of indentation on top of it being simpler to understand.
Note that I haven't tested my `std::erase_if` code so make sure you run your tests again even if you decide to take my code directly, but I know that it should work based on [this](https://en.cppreference.com/w/cpp/container/vector/erase2).

This is what your code currently looks like:
![2024-03-11_01-02_indent1](https://github.com/UE4SS-RE/RE-UE4SS/assets/91646798/44b848d3-1cdc-4435-864b-7813de482488)

This is what it looks like after I run clang-format:
![2024-03-11_01-03_indent2](https://github.com/UE4SS-RE/RE-UE4SS/assets/91646798/079d0d7f-1d8f-41bd-b987-0d4f7761635b)

This is what it looks like with erase_if and clang-format:
![2024-03-11_01-03_indent3](https://github.com/UE4SS-RE/RE-UE4SS/assets/91646798/c4bc06ec-629c-44cf-90f9-247962fb68ef)

Reply:
Will test